### ISSUE TITLE
Configurable timeout for manually stopping instances

### ISSUE THREAD

**Issue by vosdev** [lxc/lxd#9333]
# Required information

 * Distribution: Ubuntu 20.04
 * LXD version: 4.18

# Issue description

https://linuxcontainers.org/lxd/docs/master/api-extensions#container95host95shutdown95timeout

The `boot.host_shutdown_timeout` option is only used on a clean shutdown of the LXD daemon. 

I would like to request a similar option (`boot.instance_shutdown_timeout`?) that is used for the timeout when stopping instances like  `lxc stop c1`. 

Currently that timeout is only configurable through the `--timeout` option on the command itself like `lxc stop --timeout 90 c1`, with a default of 30 seconds.

Basically all my VMs with kubernetes take little over 30 seconds to shutdown so I run into this problem every time I stop one of those instances and forget the --timeout option.

**Comment by tomponline** [comment]
Or perhaps we just use the same `boot.host_shutdown_timeout` setting for `lxc stop` too if no `--timeout` flag specified.

**Comment by stgraber** [comment]
So from what I remember, `lxc stop` isn't supposed to have a timeout at all unless `--timeout` is set.

**Comment by stgraber** [comment]
```
»···// We default to -1 (i.e. no timeout) here instead of 0 (instant timeout).
»···req.Timeout = -1
»···if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
»···»···return response.BadRequest(err)
»···}
```

**Comment by stgraber** [comment]
And I confirmed that the CLI sets the timeout value to `-1` too.

**Comment by stgraber** [comment]
I've had a `lxc stop` running for over 2 minutes here so seems to be working, though that's with a container.

**Comment by stgraber** [comment]
Testing with a VM now (using Windows as it ignores the first shutdown request so makes for an easy test subject).

**Comment by tomponline** [comment]
I think there maybe be a misunderstanding here. The original post here https://discuss.linuxcontainers.org/t/boot-host-shutdown-timeout-ignored/12286/5 was expecting that `boot.host_shutdown_timeout` apply to `lxc stop` as well as `lxd shutdown`, rather than having to specify `--timeout` explicitly.

**Comment by stgraber** [comment]
@tomponline the way I read the forum post, the issue is that it's timing out after 30s when the VM takes longer than that to shutdown.

My point is that there is no such timeout in place normally, so the user shouldn't have to pass `--timeout` to get a "longer" value as the default is to wait indefinitely.

**Comment by stgraber** [comment]
```
root @ node1 # lxc stop micro1 micro2 micro3

micro1: error: Instance stop operation timed out after 30 seconds
micro1: Try `lxc info --show-log micro1` for more info
micro3: error: Instance stop operation timed out after 30 seconds
micro3: Try `lxc info --show-log micro3` for more info

Error: Some instances failed to stop
```

Is the problem that was reported in the forum post and which I'm unable to reproduce on current LXD 4.19 as my `lxc stop` correctly waits for far more than 30s.

**Comment by tomponline** [comment]
I'd need @vosdev to confirm, but my thinking was that @vosdev wanted `lxc stop` to apply a 90s timeout based on the instance's setting, rather than have no timeout or to have to specify `--timeout`.

**Comment by stgraber** [comment]
Sounds like a separate issue to me, the first thing which was showed over there is `lxc stop` failing after 30s would be a bug. As we are not setting a timeout internally, there's really nothing to bump/align with the shutdown timeout.

I also don't think we can change the behavior to timeout based on `boot.host_shutdown_timeout` as anyone who currently sets `boot.host_shutdown_timeout` but expects a manual `lxc stop` to wait longer would get a change in behavior.

We probably could add a `boot.shutdown_timeout` which would override the default of `wait indefinitely` and have `boot.host_shutdown_timeout` default to that when not set, but again, doing so right now would do nothing to fix the issue showed on the forum thread :)

**Comment by tomponline** [comment]
Agreed the example in the forum thread is strange as no `--timeout` of 30s was specified and it defaults to no timeout.

**Comment by vosdev** [comment]
> @tomponline the way I read the forum post, the issue is that it's timing out after 30s when the VM takes longer than that to shutdown.
> 
> My point is that there is no such timeout in place normally, so the user shouldn't have to pass `--timeout` to get a "longer" value as the default is to wait indefinitely.


I run LXD 4.18 with a 3 node cluster, my `lxc stop` command for these VMs times out after 30 seconds, or any other VM that is slow at shutting down. I am not specifying a `--timeout`. 

If there is no timeout by default, that's great. Then my stop operation timing out is not related to this feature request and then I'm not sure if this feature request is needed. 
The error made me think it was related.

I'll move to to the forum for support :-)

**Comment by tomponline** [comment]
Ah OK thats useful to know, I'll keep this open as it may be a different issue.
